File: CS/WpfApplication/MainViewModel.cs
﻿using DevExpress.Mvvm;
using System.Collections.ObjectModel;
using System.Linq;

namespace WpfApplication {
    public class MainViewModel : ViewModelBase {
        public ObservableCollection<Item> Items {
            get { return GetValue<ObservableCollection<Item>>(); }
            set { SetValue(value); }
        }
        public Item CurrentItem {
            get { return GetValue<Item>(); }
            set { SetValue(value); }
        }

        public MainViewModel() {
            Items = GetData();
            CurrentItem = Items.FirstOrDefault();
        }

        static ObservableCollection<Item> GetData() {
            var collection = Enumerable.Range(0, 15).Select(x => new Item { AllowEdit = x % 3 != 0, Name = $"Name{x}", ID = x });
            return new ObservableCollection<Item>(collection);
        }
    }

    public class Item : BindableBase {
        public string Name {
            get { return GetValue<string>(); }
            set { SetValue(value); }
        }
        public int ID {
            get { return GetValue<int>(); }
            set { SetValue(value); }
        }
        public bool AllowEdit {
            get { return GetValue<bool>(); }
            set { SetValue(value); }
        }
    }
}

File: VB/WpfApplication/MainViewModel.vb
Imports DevExpress.Mvvm
Imports System.Collections.ObjectModel
Imports System.Linq

Namespace WpfApplication

    Public Class MainViewModel
        Inherits ViewModelBase

        Public Property Items As ObservableCollection(Of Item)
            Get
                Return GetValue(Of ObservableCollection(Of Item))()
            End Get

            Set(ByVal value As ObservableCollection(Of Item))
                SetValue(value)
            End Set
        End Property

        Public Property CurrentItem As Item
            Get
                Return GetValue(Of Item)()
            End Get

            Set(ByVal value As Item)
                SetValue(value)
            End Set
        End Property

        Public Sub New()
            Items = GetData()
            CurrentItem = Items.FirstOrDefault()
        End Sub

        Private Shared Function GetData() As ObservableCollection(Of Item)
            Dim collection = Enumerable.Range(0, 15).[Select](Function(x) New Item With {.AllowEdit = x Mod 3 <> 0, .Name = $"Name{x}", .ID = x})
            Return New ObservableCollection(Of Item)(collection)
        End Function
    End Class

    Public Class Item
        Inherits BindableBase

        Public Property Name As String
            Get
                Return GetValue(Of String)()
            End Get

            Set(ByVal value As String)
                SetValue(value)
            End Set
        End Property

        Public Property ID As Integer
            Get
                Return GetValue(Of Integer)()
            End Get

            Set(ByVal value As Integer)
                SetValue(value)
            End Set
        End Property

        Public Property AllowEdit As Boolean
            Get
                Return GetValue(Of Boolean)()
            End Get

            Set(ByVal value As Boolean)
                SetValue(value)
            End Set
        End Property
    End Class
End Namespace
